src/pages/index.vue

```diff
@@ -51,9 +53,20 @@ const props = defineProps({
         type: String,
         required: false,
         default: ""
+    },
+    adminLevel: {
+        type: String,
+        required: false,
+        default: ""
     }
 });
 
+const getRouterLink = (name: string) => {
+    return selectedCountryId.value
+        ? `/${APP_BASE_ROUTE}/${name}/${selectedCountryId.value}/${props.adminLevel}`
+        : `/${APP_BASE_ROUTE}/${name}/`;
+};
```

Comment:
As I say, I think it would be better to default to cleaner url without admin2 part (which can optionally by provided) if there's a selected country and admin level is 2.